### PR TITLE
Data pipeline refactoring

### PR DESCRIPTION
**Pull request by hadipash**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

Refactored data pipeline to match best MindData practices, including:
1. Use `GeneratorDataset` for data loading only.
2. Use `dataset.map` operation to apply data transformations and augmentations.
3. Reduce number of Python transformations by grouping them into a single operation.
4. Group MindSpore operations as well.
5. Move to MindSpore operations where it is possible (`Decode`, `Normalize`, `HWC2CHW`).
6. Integrate MindRecord support.

### CODE REVIEW

File: mindocr/data/predict_dataset.py [hunk 1]
```diff
@@ -1,81 +1,73 @@
...
-        self.data_list = self.load_data_list(img_dir, sample_ratio, shuffle)
+        super().__init__(data_dir=data_dir)
 
-        # create transform
-        if transform_pipeline is not None:
-            self.transforms = create_transforms(transform_pipeline)  # , global_config=global_config)
-        else:
-            raise ValueError("No transform pipeline is specified!")
+        if isinstance(sample_ratio, float):
+            sample_ratio = [sample_ratio] * len(self.data_dir)
```
Comment (hadipash) [inline mindocr/data/predict_dataset.py:43]: I don't think `sample_ratio` is needed in `PredictDataset`.

File: mindocr/data/predict_dataset.py [hunk 2]
```diff
@@ -1,81 +1,73 @@
...
         self,
-        # is_train: bool = False,
-        dataset_root: str = "",
-        data_dir: str = "",
+        data_dir: Union[str, List[str]] = None,
         sample_ratio: Union[List, float] = 1.0,
-        shuffle: bool = None,
-        transform_pipeline: List[dict] = None,
-        output_columns: List[str] = None,
+        shuffle: bool = False,
```
Comment (hadipash) [inline mindocr/data/predict_dataset.py:37]: I think `sample_ratio` and `shuffle` are not needed in `PredictDataset`.

File: mindocr/data/rec_lmdb_dataset.py [hunk 3]
```diff
@@ -15,20 +14,16 @@ class LMDBDataset(BaseDataset):
...
-        data_dir: data root directory for lmdb dataset(s)
-        shuffle: Optional, if not given, shuffle = is_train
-        transform_pipeline: list of dict, key - transform class name, value - a dict of param config.
-                    e.g., [{'DecodeImage': {'img_mode': 'BGR', 'channel_first': False}}]
-            -       if None, default transform pipeline for text detection will be taken.
-        output_columns (list): optional, indicates the keys in data dict that are expected to output for dataloader.
-            if None, all data keys will be used for return.
-        filter_max_len (bool): Filter the records where the label is longer than the `max_text_len`.
-        max_text_len (int): The maximum text length the dataloader expected.
+        is_train: Whether the dataset is for training. Default: True.
```
Comment (hadipash) [inline mindocr/data/rec_lmdb_dataset.py:17]: @zhtmike we can remove `is_train` and set `shuffle` to False by default. There's no reason to shuffle training set on initialization if the loader shuffles it on every iteration. We can keep the `shuffle` parameter for cases when a training set is partially used (`sample_ratio` < 1).

Comment (zhtmike) [reply mindocr/data/rec_lmdb_dataset.py:17]: Agree, we can remove it in the next PR

File: mindocr/data/predict_dataset.py (hunk 2 above)
Comment (HaoyangLee) [reply mindocr/data/predict_dataset.py]: `shuffle` is removed. `sample_ratio` is kept, for the case that users want to quickly see the inference results for checking whether his model is good but there are too much images in his dataset folder.

**Comment by hadipash** [comment]
Rebase onto the `main` branch to resolve conflicts.

**Review by SamitHuang — APPROVED** [review]
Nice work. Training performance for the two detection models is greatly improved.

**Pull request closed without merging**